webui/src/components/editfiles.jsx

```diff
@@ -266,6 +266,10 @@ export const EditFiles = React.createClass({
         }
     },
 
+    componentWillReceiveProps(props) {
+        return props.readOnly !== this.props.readOnly;
```

Comment:
I'm not sure what this does, I can't find any docs about a return value from `componentWillReceiveProps`, can you help?

Reply:
Right, I was going to use `shouldComponentUpdate` but there were some issue that I can't recall now, so I decided to use `componentWillReceiveProps` instead and some how got confused! I'll check it again.